src/vm_cpu.rs

```diff
@@ -340,6 +399,12 @@ impl<'vcpu, 'pages, 'prev, T: GuestStagePagingMode> ActiveVmCpu<'vcpu, 'pages, '
 
         // TODO: Enforce that the vCPU has an assigned interrupt file before running.
 
+        let has_vector = CpuInfo::get().has_vector();
+
+        if has_vector {
+            run_guest_vector(&mut self.state as *mut VmCpuState);
```

Comment:
nit: No need for the cast, here and below.

Reply:
will fix